app/models/repository.rb

```diff
@@ -6,4 +6,12 @@ class Repository < ApplicationRecord
   def to_s
     link
   end
+
+  scope :languages, -> { select(:language).distinct.where.not(language: [nil, '']) }
```

Comment:
Why filter only to the language here?

if you want repositories with languages what about `Repository.with_languages`

Reply:
That scope is being used on repositories languages filter, which shows the available distinct languages from the repositories. Indeed that scope's name's not clear enough as it should be, on my way to change that

Reply:
I got it; my question was about the `select(:language)` you just want the the laguages,

so there are two problems here

* Scope is not the correct approach; the scope is used to filter (scope) the Repositories, and here you want an array of languages
* Te second problem is that the language column, can have multiple languages, it can have "Ruby", "Javascript" and can also have "Ruby, Javascrit", "Javascript, Ruby", "Ruby, React, javascript", etc ...